python/example_code/s3/s3_versioning/versioning.py

```diff
@@ -0,0 +1,259 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Purpose
+
+Demonstrates how to set up an Amazon S3 bucket for versioning, and how to perform
+basic tasks on a version-enabled bucket.
+
+For more detail, including usage and testing instructions, see the README.
+
+Running this demo uses your default AWS credentials to create resources in your
+account and my incur charges.
+"""
+
+import logging
+from operator import attrgetter
+from shutil import get_terminal_size
+from sys import stdout
+import uuid
+
+import boto3
+from botocore.exceptions import ClientError
+
+logger = logging.getLogger(__name__)
+log_handler = logging.StreamHandler(stdout)
+log_handler.setLevel(logging.INFO)
+logger.addHandler(log_handler)
+logger.setLevel(logging.INFO)
```

Comment:
Slightly cleaner way to write all this might be:

```
logging.basicConfig(level=os.environ.get('LOGLEVEL', 'INFO').upper(), stream=sys.stdout)
boto3.set_stream_logger('', logging.INFO)
```

Reply:
Good suggestion. I took it and modified it slightly.